File: web/src/lib/data-normalizer.ts
/**
 * Data normalizer for restaurant data.
 * Handles both English enum values (from normalized backend) and
 * legacy Hebrew values (backward compatibility).
 */

const PRICE_MAP: Record<string, string> = {
  'budget': 'budget', 'mid-range': 'mid-range', 'expensive': 'expensive',
  'זול': 'budget', 'תקציבי': 'budget',
  'בינוני': 'mid-range', 'בינוני-יקר': 'mid-range',
  'יקר': 'expensive', 'יוקרתי': 'expensive',
};

const STATUS_MAP: Record<string, string> = {
  'open': 'open', 'closed': 'closed', 'new_opening': 'new_opening',
  'closing_soon': 'closing_soon', 'reopening': 'reopening',
  'פתוח': 'open', 'סגור': 'closed',
  'חדש': 'new_opening', 'פתיחה חדשה': 'new_opening',
  'נסגר בקרוב': 'closing_soon', 'נפתח מחדש': 'reopening',
};

const OPINION_MAP: Record<string, string> = {
  'positive': 'positive', 'negative': 'negative', 'mixed': 'mixed', 'neutral': 'neutral',
  'חיובית מאוד': 'positive', 'חיובית': 'positive',
  'שלילית': 'negative', 'מעורבת': 'mixed',
  'ניטרלית': 'neutral', 'לא צוין': 'neutral',
};

export function normalizePriceRange(value?: string | null): string | null {
  if (!value) return null;
  return PRICE_MAP[value] ?? null;
}

export function normalizeStatus(value?: string | null): string | null {
  if (!value) return null;
  return STATUS_MAP[value] ?? value;
}

export function normalizeHostOpinion(value?: string | null): string | null {
  if (!value) return null;
  return OPINION_MAP[value] ?? value;
}

export interface MenuItem {
  item_name: string;
  description?: string;
  price?: string | null;
  recommendation_level?: string;
}

export function normalizeMenuItems(items?: unknown[]): MenuItem[] {
  if (!items || items.length === 0) return [];
  return items.map(item => {
    if (typeof item === 'string') {
      return { item_name: item, description: '', price: null, recommendation_level: 'mentioned' };
    }
    return item as MenuItem;
  });
}

export function getPriceDisplay(priceRange?: string | null): string {
  const normalized = normalizePriceRange(priceRange);
  switch (normalized) {
    case 'budget': return '₪';
    case 'mid-range': return '₪₪';
    case 'expensive': return '₪₪₪';
    default: return '';
  }
}

File: src/data_normalizer.py
"""
Data normalizer for Where2Eat restaurant data.
Converts Hebrew AI analyzer output values to English enum values
expected by the frontend TypeScript types.
"""

PRICE_RANGE_MAP = {
    'זול': 'budget', 'תקציבי': 'budget',
    'בינוני': 'mid-range', 'בינוני-יקר': 'mid-range',
    'יקר': 'expensive', 'יוקרתי': 'expensive',
    'לא צוין': 'not_mentioned',
    # English passthrough
    'budget': 'budget', 'mid-range': 'mid-range', 'expensive': 'expensive',
    'not_mentioned': 'not_mentioned',
}

STATUS_MAP = {
    'פתוח': 'open', 'open': 'open',
    'סגור': 'closed', 'closed': 'closed',
    'חדש': 'new_opening', 'פתיחה חדשה': 'new_opening', 'new_opening': 'new_opening',
    'נסגר בקרוב': 'closing_soon', 'closing_soon': 'closing_soon',
    'נפתח מחדש': 'reopening', 'reopening': 'reopening',
    'לא צוין': None,
}

OPINION_MAP = {
    'חיובית מאוד': 'positive', 'חיובית': 'positive', 'positive': 'positive',
    'שלילית': 'negative', 'negative': 'negative',
    'מעורבת': 'mixed', 'mixed': 'mixed',
    'ניטרלית': 'neutral', 'neutral': 'neutral',
    'לא צוין': 'neutral',
}


def normalize_price_range(value):
    if not value:
        return None
    return PRICE_RANGE_MAP.get(value, value)


def normalize_status(value):
    if not value:
        return None
    return STATUS_MAP.get(value, value)


def normalize_host_opinion(value):
    if not value:
        return None
    return OPINION_MAP.get(value, value)


def normalize_menu_items(items):
    """Convert string items to MenuItem objects."""
    if not items:
        return []
    result = []
    for item in items:
        if isinstance(item, str):
            result.append({
                'item_name': item,
                'description': '',
                'price': None,
                'recommendation_level': 'mentioned'
            })
        elif isinstance(item, dict):
            result.append(item)
    return result


def normalize_restaurant(data):
    """Normalize all fields of a restaurant dict."""
    data['price_range'] = normalize_price_range(data.get('price_range'))
    data['status'] = normalize_status(data.get('status'))
    data['host_opinion'] = normalize_host_opinion(data.get('host_opinion'))
    data['menu_items'] = normalize_menu_items(data.get('menu_items', []))
    # Remove photo_url from photos (use photo_reference via proxy)
    for photo in data.get('photos', []):
        photo.pop('photo_url', None)
    return data
